webapp/src/components/Modals/ClaimNameFatFingerModal/ClaimNameFatFingerModal.tsx

```diff
@@ -17,56 +36,189 @@ export const CONTROLLER_V2_ADDRESS = config.get(
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus()
+    }
+  }, [])
+
+  const [paymentMethod, setPaymentMethod] = useState(
+    !!(wallet && currentMana && isEnoughClaimMana(currentMana))
+      ? PaymentMethod.CRYPTO
+      : isClaimingNamesWithFiatEnabled
+      ? PaymentMethod.FIAT
+      : PaymentMethod.CRYPTO
+  )
   const [currentName, setCurrentName] = useState('')
 
-  const handleClaim = useCallback(() => {
-    const mana = getContract({
-      name: getContractNames().MANA,
-      network: Network.ETHEREUM
-    })
+  const handleClaimWithCard = useCallback(async () => {
+    const wertURL = config.get('WERT_URL')
+    if (wallet) {
+      const signer = await getSigner()
+      const factory = await DCLController__factory.connect(
+        CONTROLLER_V2_ADDRESS,
+        signer
+      )
+
+      const sc_input_data = factory.interface.encodeFunctionData('register', [
+        ENSName,
+        wallet.address
+      ])
+
+      const data = {
+        address: wallet.address,
+        commodity: isDev ? 'TTS' : 'MANA', // will be MANA later on
+        commodity_amount: Number(PRICE),
+        network: isDev ? 'sepolia' : 'ethereum', // will be wallet.network
+        sc_address: '0x39421866645065c8d53e2d36906946f33465743d',
+        sc_input_data
+      }
 
-    if (!mana) return
+      const nftOptions = {
+        extra: {
+          item_info: {
+            category: 'Decentraland NAME',
+            author: 'Decentraland',
+            image_url: `${MARKETPLACE_SERVER_URL}/ens/generate?ens=${ENSName}&width=330&height=330`,
+            ENSName,
+            seller: 'DCL Names'
+          }
+        }
+      }
 
-    const manaContract = {
-      name: ContractName.MANAToken,
-      address: mana.address,
-      chainId: getChainIdByNetwork(Network.ETHEREUM),
-      network: Network.ETHEREUM,
-      category: NFTCategory.ENS
+      onOpenFiatGateway(
+        FiatGateway.WERT,
+        {
+          ...data,
+          ...nftOptions,
+          partner_id: '01HGFWXR5CQMYHYSR9KVTKWDT5',
+          origin: wertURL,
+          lang: 'en',
+          click_id: uuidv4() // unique id of purchase in your system
+        },
+        {
+          onLoaded: () => {
+            setIsLoadingFIATWidget(false)
+          },
+          onSuccess: options => {
+            if ('tx_id' in options) {
+              analytics.track('Buy Name Success', {
+                name: ENSName,
+                payment_method: 'fiat',
+                txHash: options.tx_id
+              })
+              onClaimTxSubmitted(
+                ENSName,
+                wallet.address,
+                wallet.chainId,
+                options.tx_id as string
+              )
+            }
+          }
+        }
+      )
     }
+  }, [wallet, ENSName, analytics, onOpenFiatGateway, onClaimTxSubmitted])
+
+  const isLoading = useMemo(() => isClaiming || isLoadingFIATWidget, [])
```

Comment:
If I'm not wrong, as there's no dependencies in this `useMemo`, `isLoading` will be computed only once. As this is a boolean comparison, what do you think about removing the `useMemo` hook?
```suggestion
  const isLoading = isClaiming || isLoadingFIATWidget
```